lib/widgets/create_post_popup_window.dart

```diff
@@ -1,120 +1,213 @@
+import 'dart:io';
 import 'package:flutter/material.dart';
-import 'package:flutter_svg/svg.dart';
-import 'package:twitter_clone/utils/colors.dart';
+import 'package:image_picker/image_picker.dart';
 
-class CreatePostPopupWindow extends StatelessWidget {
+class CreatePostPopupWindow extends StatefulWidget {
   final double screenHeight;
 
-  const CreatePostPopupWindow({super.key, required this.screenHeight});
+  const CreatePostPopupWindow({Key? key, required this.screenHeight}) : super(key: key);
+
+  @override
+  _CreatePostPopupWindowState createState() => _CreatePostPopupWindowState();
+}
+
+class _CreatePostPopupWindowState extends State<CreatePostPopupWindow> {
+  String? _pickedImagePath;
+  final ImagePicker _picker = ImagePicker();
+
+  // Callback function when selecting an image
+  void _pickImage() async {
+    final XFile? pickedFile = await _picker.pickImage(source: ImageSource.gallery);
+    if (pickedFile != null) {
+      setState(() {
+        _pickedImagePath = pickedFile.path;
+      });
+    }
+  }
+
+  // Callback function when selecting a video
```

Comment:
We should remove this comment description because your method is quite clear and readability so we do not need it in this case because we still understand it without the comment